src/snap_storage.test.ts

```diff
@@ -120,4 +120,33 @@ describe("Snapshot storage", () => {
 
     assert(!storedSnap.isFresh, "Existing snapshots should not be fresh");
   });
+
+  it("caches a modified fetched JSON response", async () => {
+    const testFile = "test_data/mutator.json";
+    await Deno.writeTextFile(testFile, JSON.stringify({}));
+
+    const url = toFileUrl(resolve(testFile));
+    async function responseMutator(response: Response) {
+      const data = await response.json();
+      data.mutation = true;
+      return new Response(JSON.stringify(data), response);
+    }
+    const fetchedSnap = await snaps.cache(url, {
+      responseMutator: responseMutator,
+    });
+    const storedSnap = await snaps.loadJSON(url.href);
+    const fetchedJSON = await fetchedSnap.content.json();
+
+    assert(fetchedJSON.mutation, "JSON response was not mutated");
+
+    assertEquals(
+      fetchedJSON,
+      storedSnap.content,
+      "Fetched mutated JSON response has not been persisted correctly",
+    );
+
+    assert(fetchedSnap.isFresh, "Fetched snapshots have to be marked as fresh");
+
+    assert(!storedSnap.isFresh, "Existing snapshots should not be fresh");
```

Comment:
```suggestion
    const fetchedJSON = await fetchedSnap.content.json();

    assert(fetchedJSON.mutation, "JSON response was not mutated");
```
You can remove the last three assertions, they should already be covered by the previous test case.